frame/sudo/src/tests.rs

```diff
@@ -0,0 +1,67 @@
+use super::*;
+use mock::{
+	Sudo, Origin, Call, Test, PrivelegedCall, new_test_ext,
+}; 
+use frame_support::{assert_ok, assert_err};
+
+#[test]
+fn new_test_ext_and_sudo_get_key_works() {
+	// Test that the enivroment setup and pallets key retrieval works as expected
+	new_test_ext(1).execute_with(|| {
+		assert_eq!(Sudo::key(),  1u64);
+	});
+}
+
+#[test]
+fn sudo_basics() {
+	new_test_ext(1).execute_with(|| {
+		// A privelleged function should work when sudo is passed the root_key as origin
+		let call = Box::new(Call::Priveleged(PrivelegedCall::privileged_function()));
+		assert_ok!(Sudo::sudo(Origin::signed(1), call));
+		// A privelleged function should not work when called with a non-root user
+		let call = Box::new(Call::Priveleged(PrivelegedCall::privileged_function()));
+		assert_err!(Sudo::sudo(Origin::signed(2), call), Error::<Test>::RequireSudo);
+	});
+}
+
+#[test]
+fn sudo_unchecked_weight_basics() {
```

Comment:
You need to be asserting the functionality of the extrinsic you are testing.

Making a privileged call is one, the other is controlling the dispatch weight.

You should see what `dispatch_info().weight` is of a `sudo_unchecked_weight` function is.

Reply:
A call for ```sudo_unchecked_weight``` was added on line 62. ```get_dispatch_info()``` is then used on that call and saved to a variable, ```info```. ```info.weight``` is then asserted to be equal to the passed ```weight``` on line 76.